### PR TITLE
Query and test changes

### PR DESCRIPTION
**Pull request by mabe13**
This change will update sql for checking a dos user can edit a service by also checking they are an active user, and the parent service have active statuses. It also checks that any parent service that links a service to be edit to the given user is also active.

### CODE REVIEW

**Comment by mabe13** [comment]
> The additional checks themselves look fine, but should we be distinguishing between the three types of check that we have here? At the moment, it looks like we raise the 403 - DoS User Forbidden to update service, if either: a. the user is not active, b. the service is not active, or c. the user doesn't have edit permissions for the particular service. It may be clearer if we treat each case separately, and raise a specific error for each. There are a couple of ways we can do this. We can either implement each check separately, raising a more specific 403 response for each; or we can leave the check as it is, and then if we hit the 403, we can look into seeing which one of the three cases we are dealing with. What do you think?

Agree, added one commit to interogate why a user can edit a particular service and return an appropriate http error code response. There is also a second to removing checking that a parent service is active, as this isn't expected by DoS.

**Comment by mabe13** [comment]
Just added a new commit to include validating the user and service exist, and are active for GET requests, also includes returning appropriate responses (http codes) when they aren't.

File: application/api/capacityservice/views.py [hunk 1]
```diff
@@ -80,15 +85,59 @@ def patch(self, request, service__uid, partial=True):
         logger.info("Payload: %s", request.data)
         return self._process_service_status_update(request, service__uid)
 
+    def _handle_check_dos_user(self, api_key):
+        user = get_user_for_key(api_key)
+        if user is None:
+            is_valid_user = False
+            status_msg = "Given DoS user does not exists"
```
Comment (jope1) [inline application/api/capacityservice/views.py]: Typo on 'exists'

File: application/api/capacityservice/views.py [hunk 2]
```diff
@@ -80,15 +100,82 @@ def patch(self, request, service__uid, partial=True):
         logger.info("Payload: %s", request.data)
         return self._process_service_status_update(request, service__uid)
 
+    def _handle_check_dos_user(self, request):
+        api_key = self.get_permissions()[0].get_key_model(request)
+        user = get_user_for_key(api_key)
```
[the remark this answers is not in the document]
Comment (mabe13) [reply application/api/capacityservice/views.py]: Agree, resolved in live code session.

File: application/api/capacityservice/views.py [hunk 3]
```diff
@@ -80,15 +100,82 @@ def patch(self, request, service__uid, partial=True):
...
+        if user is None:
+            is_valid_user = False
+            status_msg = "Given DoS user does not exist"
+            status_code = status.HTTP_401_UNAUTHORIZED
```
[the remark this answers is not in the document]
Comment (mabe13) [reply application/api/capacityservice/views.py]: Disccussed and Solution agreed in live code session.

File: application/api/capacityservice/views.py [hunk 4]
```diff
@@ -80,15 +100,82 @@ def patch(self, request, service__uid, partial=True):
...
+    def _get_service_or_none_from_capacity(self, service_capacity):
+        if service_capacity:
+            return service_capacity.service
+        return None
+
+    def _handle_check_service(self, service):
+        if service is None:
+            is_valid_service = False
+            status_msg = "Given service does not exist"
+            status_code = status.HTTP_404_NOT_FOUND
```
[the remark this answers is not in the document]
Comment (mabe13) [reply application/api/capacityservice/views.py]: Disccussed and solution agreed in live code session.

[the remark this answers is not in the document]
Comment (mabe13) [reply application/api/capacityservice/views.py]: Disccussed and solution agreed in live code session.

File: application/api/capacityservice/views.py [hunk 5]
```diff
@@ -40,7 +45,14 @@ class CapacityStatusView(RetrieveUpdateAPIView):
     )
     def get(self, request, service__uid):
         logger.info("Request sent from host: %s", request.META["HTTP_HOST"])
-        return self._process_service_status_retrieval(request, service__uid)
+        is_user_valid, msg, sc = self._handle_check_dos_user(request)
+        if is_user_valid:
+            service_capacity = self._get_service_capcitystatus(service__uid)
```
[the remark this answers is not in the document]
Comment (mabe13) [reply application/api/capacityservice/views.py]: Agree, will change in next commit.

**Pull request merged**